node/beacon-chain-handler/src/producer.rs

```diff
@@ -66,37 +66,37 @@ impl BlockProducer {
     }
 }
 
-pub type ShardChainPayload = (Vec<SignedTransaction>, Vec<ReceiptTransaction>);
+pub type ShardChainPayload = Vec<Transaction>;
 pub type ChainConsensusBlockBody = ConsensusBlockBody<ShardChainPayload>;
 
 impl ConsensusHandler<SignedBeaconBlock, ShardChainPayload> for BlockProducer {
     fn produce_block(&self, body: ChainConsensusBlockBody) {
         // TODO: verify signature
         let transactions = body.messages.iter()
-            .flat_map(|message| message.body.payload.0.clone())
-            .collect();
-        let receipts = body.messages.iter()
-            .flat_map(|message| message.body.payload.1.clone())
+            .flat_map(|message| message.body.payload.clone())
             .collect();
 
         // TODO: compute actual merkle root and state, as well as signature, and
```

Comment:
remove this todo - fork choice rule does not apply when block is produced. and the rest of stuff is already done.

Reply:
Fixed in 4bed468